public/app/mu-plugins/moj-auth/401.php

```diff
@@ -71,23 +71,33 @@ public function handle401Request(): void
     {
         $this->log('handle401Request()');
 
+        // Return early if it's a heartbeat request. 
+        // If that endpoint is a 401 do nothing here.
+        if ($_SERVER['REQUEST_URI'] === '/auth/heartbeat') {
+            return;
+        }
+
         // Get the JWT token from the request. Do this early so that we populate $this->sub if it's known.
         $jwt = $this->getJwt() ?: (object)[];
 
-        // Set loginAttempts with a default of 1, or add one to the existing value.
-        $jwt->login_attempts = empty($jwt->login_attempts) ? 1 : ((int) $jwt->login_attempts) + 1;
+        $this->log('Request URI: ' . $_SERVER['REQUEST_URI']);
 
         // Always add the schema and domain here, to prevent an open redirect vulnerability.
         $jwt->success_url = $_ENV['WP_HOME'] . $_SERVER['REQUEST_URI'];
 
         // Set the cookie expiry to 0 to create a session cookie.
         $jwt->cookie_expiry = 0;
 
+        // Set failed_callbacks with a default of 0.
+        $jwt->failed_callbacks = isset($jwt->failed_callbacks) ? $jwt->failed_callbacks : 0;
```

Comment:
This might be better, null coalesce:

```php
$jwt->failed_callbacks = $jwt->failed_callbacks ?? 0;
```

Where is `$failed_callbacks` being declared, is it in a trait or class? 
It looks like the var is being dynamically set.

Reply:
Good shout.

On the usual login flow, this will be the first place that `$jwt->failed_callbacks` is set.